pkg/cmd/server/origin/controller/network.go

```diff
@@ -0,0 +1,68 @@
+package controller
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	osclient "github.com/openshift/origin/pkg/client"
+	configapi "github.com/openshift/origin/pkg/cmd/server/api"
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+	sdnplugin "github.com/openshift/origin/pkg/sdn/plugin"
+	"github.com/openshift/origin/pkg/service/controller/ingressip"
+)
+
+type SDNControllerConfig struct {
+	NetworkConfig configapi.MasterNetworkConfig
+}
+
+func (c *SDNControllerConfig) RunController(ctx ControllerContext) (bool, error) {
+	// TODO: Switch SDN to use client.Interface
```

Comment:
let's get issues open for these.  Individual issues will help us distribute the pain.